things/postgres/things.go

```diff
@@ -160,36 +177,50 @@ func (tr thingRepository) RetrieveByChannel(owner, channel string, offset, limit
 	      FROM things th
 	      INNER JOIN connections co
 		  ON th.id = co.thing_id
-		  WHERE th.owner = $1 AND co.channel_id = $2
+		  WHERE th.owner = :owner AND co.channel_id = :channel
 		  ORDER BY th.id
-		  LIMIT $3
-		  OFFSET $4`
-	items := []things.Thing{}
+		  LIMIT :limit
+		  OFFSET :offset;`
+
+	params := map[string]interface{}{
+		"owner":   owner,
+		"channel": channel,
+		"limit":   limit,
+		"offset":  offset,
+	}
 
-	rows, err := tr.db.Query(q, owner, channel, limit, offset)
+	rows, err := tr.db.NamedQuery(q, params)
 	if err != nil {
 		tr.log.Error(fmt.Sprintf("Failed to retrieve things due to %s", err))
 		return things.ThingsPage{}
 	}
 	defer rows.Close()
 
+	items := []things.Thing{}
 	for rows.Next() {
-		t := things.Thing{Owner: owner}
-		if err := rows.Scan(&t.ID, &t.Type, &t.Name, &t.Key, &t.Metadata); err != nil {
+		dbth := dbThing{Owner: owner}
+		if err := rows.StructScan(&dbth); err != nil {
+			tr.log.Error(fmt.Sprintf("Failed to read retrieved thing due to %s", err))
+			return things.ThingsPage{}
+		}
+
+		th, err := toThing(dbth)
+		if err != nil {
 			tr.log.Error(fmt.Sprintf("Failed to read retrieved thing due to %s", err))
 			return things.ThingsPage{}
 		}
-		items = append(items, t)
+
+		items = append(items, th)
 	}
 
 	q = `SELECT COUNT(*)
 	     FROM things th
 	     INNER JOIN connections co
 	     ON th.id = co.thing_id
-	     WHERE th.owner = $1 AND co.channel_id = $2`
+	     WHERE th.owner = $1 AND co.channel_id = $2;`
```

Comment:
It's normal that here you don't use `th.owner = :owner AND co.channel_id = :channel` ?

Reply:
Because I used that that doesn't support named parameters. There are other cases where I haven't used named parameters. If I've called `NamedExec` or `NamedQuery`, then I can use named parameters, otherwise I can't.

Reply:
Ok I get it. Resolving this one